### PR TITLE
Removing stanford_opportuntities ct

### PR DESCRIPTION
**Pull request by mdyoung3**
# READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- Removes duplicate opportunities on post update that stanford profiles adds. 

# Review By (Date)
- October 7

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Urgency
- How urgent is this? (Normal, High)

# Review Tasks

## Setup tasks and/or behavior to test

1. Reviews code

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: cardinal_service_profile.post_update.php [hunk 1]
```diff
@@ -125,3 +125,14 @@ function cardinal_service_profile_post_update_site_orgs() {
     }
   }
 }
+
+/**
+ * Removes stanford opportunity content type created by Stanford Profile.
+ * To prevent confusion between similarly named CT already configured.
+ */
+function cardinal_service_profile_post_update_remove_stanford_opportunities() {
```
Comment (pookmish) [inline cardinal_service_profile.post_update.php:133]: Why is this needed? the update hook won't actually do anything since the configuration is still there. Why not just change the name. Permissions should prevent the access to the content type anyways.

**Review by pookmish — CHANGES_REQUESTED** [review]

File: cardinal_service_profile.post_update.php (hunk 1 above)
Comment (mdyoung3) [reply cardinal_service_profile.post_update.php:133]: @pookmish There are two Opportunity content types on the latest version of Cardinal Service: https://cardinalservice-dev.sites-pro.stanford.edu/admin/structure/types

Judging by the stanford profile, version 11.4.0 add an Opportunity content type to the profile: https://github.com/SU-SWS/stanford_profile/blob/11.x/CHANGELOG.md 

I could be wrong, but won't it come back since it's part of the Stanford Profile? Hence I added the post-update removal...

Comment (mdyoung3) [reply cardinal_service_profile.post_update.php:133]: Ooo. It does look like cardinal service profile is tracking it. Maybe I need to add it to the config ignore as well.

Comment (pookmish) [reply cardinal_service_profile.post_update.php:133]: That's why i suggest just changing the name of each content type. that way it won't break inheritance issues. Config ignore won't do anything. The site managers don't see it anyways except when filtering content on the /admin/content page.

Comment (mdyoung3) [reply cardinal_service_profile.post_update.php:133]: I did a masquerade and you're right they won't see. But won't renaming it just be overridden the next update?

Comment (pookmish) [reply cardinal_service_profile.post_update.php:133]: It will only cause merge conflicts if both the local line and the remote line change. So if we change the label in this repo AND stanford_profile changes the label, it will cause a merge conflict... which we just resolve like normal.

Comment (mdyoung3) [reply cardinal_service_profile.post_update.php:133]: I can see conflict merges being a pain point for future devs but I don't have a better solution. So, I'll close this PR and get another one going.
